Modules/posixmodule.c

```diff
@@ -5467,6 +5467,50 @@ os__path_islink_impl(PyObject *module, PyObject *path)
 #endif /* MS_WINDOWS */
 
 
+/*[clinic input]
+os._path_splitroot_ex
+
+    path: unicode
+
+[clinic start generated code]*/
+
+static PyObject *
+os__path_splitroot_ex_impl(PyObject *module, PyObject *path)
+/*[clinic end generated code: output=de97403d3dfebc40 input=f1470e12d899f9ac]*/
+{
+    Py_ssize_t len, drvsize, rootsize;
+    PyObject *drv = NULL, *root = NULL, *tail = NULL, *result = NULL;
+
+    wchar_t *buffer = PyUnicode_AsWideCharString(path, &len);
+    if (!buffer) {
+        goto exit;
+    }
+
+    _Py_skiproot(buffer, len, &drvsize, &rootsize);
+    if (!(drv = PyUnicode_FromWideChar(buffer, drvsize)) ||
+        !(root = PyUnicode_FromWideChar(&buffer[drvsize], rootsize)) ||
+        !(tail = PyUnicode_FromWideChar(&buffer[drvsize + rootsize], len - drvsize - rootsize)))
+    {
+        goto exit;
+    }
```

Comment:
This C style is not commonly used in our code base; consider spelling it out:

```suggestion
    drv = PyUnicode_FromWideChar(buffer, drvsize);
    if (drv == NULL) {
        goto exit;
    }
    root = PyUnicode_FromWideChar(&buffer[drvsize], rootsize);
    if (root == NULL) {
        goto exit;
    }
    tail = PyUnicode_FromWideChar(&buffer[drvsize + rootsize],
                                  len - drvsize - rootsize);
    if (tail == NULL) {
        goto exit;
    }
```

Reply:
How uncommon is 525 times? Regex: `if \(.* = `. Or do you mean the multiple conditions?
And I think spelling it out makes it less readable, but you have the final say.

Reply:
@nineteendo, I'd appreciate if you could apply this suggestion as well. We try to keep our C code as readable as possible.

Reply:
> How uncommon is 525 times?

The correct number is much lower[^1], and it really does not matter since it is still the minority.

[^1]: for example, mimalloc and libmpdec are vendored, Python-ast.c is autogenerated.

Reply:
I see, then I'll stick to the guidelines.